fix(K8s): Merged reduce data ingest and data governance pages (#17610)
newrelic/docs-website: src/content/docs/kubernetes-pixie/kubernetes-integration/installation/reduce-ingest.mdx

---
title: Reduce data ingested
tags:
    - Integrations
    - Kubernetes integration
    - Understand and use data
metaDescription: 'New Relic''s Kubernetes integration: How to reduce the data ingested.'
redirects:
    - /docs/integrations/kubernetes-integration/understand-use-data/find-use-your-kubernetes-data
    - /docs/integrations/kubernetes-integration/understand-use-data/understand-use-data
    - /docs/integrations/kubernetes-integration/installation/install-kubernetes-integration-using-helm/#services
freshnessValidatedDate: 2024-06-24
---

> If you've checked your data ingestion and you'd like to reduce the amount you have, you can do it through charts and through parameters.

Reviewer: I changed a little bit the content, because those settings are related only to Helm Chart. If you want to change directly in a Manifest YAML file is more complex, for example low Data Mode parameter in the Helm Chart does several changes in the final Manifest YAML file applied.

Here is my proposal:

If you've reviewed your data ingestion (https://docs.newrelic.com/docs/data-apis/manage-data/manage-data-coming-new-relic/) and want to reduce the volume of data you're sending, we've got a simple solution for you. You can easily achieve this by setting up our nri-bundle (https://github.com/newrelic/helm-charts/tree/master/charts/nri-bundle) Helm chart. Our charts provide a way to fine-tune your settings and reduce the amount of data ingested, which may result in slightly less detailed information. These setting, also allow you to focus on the most critical data points, helping you identify the most important insights

Author: Done!

You can reduce the data in your charts and works for all installation methods: the guided install, Helm, or Manifest. Our charts can help you adjust settings to reduce the data you take in, which may result in less detailed information. These settings allow you to focus on the most important data, which can help you identify the most important insights.

It's also possible to reduce data ingestion through parameters. However, the modification of the parameters will depend on the method used to install Kubernetes.

## Reduce data ingested by setting `global.lowDataMode` [#reduce-data-ingest]

If you're looking to cut costs, you can modify the `global.lowDataMode` key value. This will cut down on the number of metrics sent. The modification of this key will depend on the method used to install Kubernetes:

>     * **Guided install** (New Relic CLI, Helm, Manifest). The`global.lowDataMode` key is enabled by default.

Reviewer: Guided install (New Relic CLI, Helm, Manifest).  The lowDataMode mode is included by default in the command thanks to 3 parameters (NRI_CLI_LOW_DATA_MODE, NRI_CLI_PROMETHEUS_AGENT_LOW_DATA_MODE and NRI_CLI_LOGGING_LOW_DATA_MODE).

Author: Done.

    * **Helm**. Set the `global.lowDataMode` key in the [`nri-bundle` chart](https://github.com/newrelic/helm-charts/tree/master/charts/nri-bundle) to true.

>     * **Manifest**. Set the `global.lowDataMode` key in the Manifest configuration yaml file to true. If you followed our [installation instructions](/install/kubernetes/?dropdown1=manifest), the file was called `newrelic-manifest.yaml`.

Reviewer: I would remove this or entirely changes.
LowDataMode is exclusive to Helm Charts. Helms are templates applied to Manifest, when someone applies the lowDataMode in the Helm Chart it applies several small changes in the final Manifest.
Changes applied in the Manifest are about specific settings but any of them is about a parameter named lowDataMode.

Author: Added your paragraph.

> `lowDataMode` key affects these specific components of the `nri-bundle` chart:

Reviewer: Can we include a link to the nri-bundle Helm Chart?
https://github.com/newrelic/helm-charts/tree/master/charts/nri-bundle

Author: Sure! Added.

    }
    ]
    ```

    This is how the previous log record would look like after enabling `lowDataMode`:

    ```json
    [
    {
        "cluster_name": "api-test",
        "container_name": "newrelic-logging",
        "namespace_name": "nrlogs",
        "pod_name": "nri-bundle-newrelic-logging-jxnbj",
        "message": "[2021/09/14 12:30:49] [ info] [engine] started (pid=1)\n",
        "stream": "stderr",
        "timestamp": 1631622649138
    }
    ]
    ```
    </Collapser>
    <Collapser
        id="nr-pixie-integration"
        title="New Relic Pixie integration"
    >

    If you enable `lowDataMode`, the `newrelic-pixie` integration performs heavier sampling on Pixie spans and collect data less often, from 10 seconds to 15 seconds. These are the `lowDataMode` settings:

    ```
    HTTP_SPAN_LIMIT: 750
    DB_SPAN_LIMIT: 250
    COLLECT_INTERVAL_SEC: 15
    ```

    You can find the default settings for these parameters and others in the [newrelic-pixie-integration](https://github.com/newrelic/newrelic-pixie-integration) Github repo.

    </Collapser>
</CollapserGroup>

## Reduce data ingested by changing the scrape interval [#scrape-interval]

> The New Relic Kubernetes integration allows changing the interval at which metrics are gathered from the cluster. This lets you choose the right balance between data resolution and usage. We suggest choosing an interval between 15 and 30 seconds for the best experience.

Reviewer: We should comment that lowDataMode already sets the scrape interval to 30s

Author: Added a tip note.

The modification of this interval will depend on the method used to install Kubernetes:

    * **Guided install** (New Relic CLI, Helm, Manifest). You can't modify this value following our guided install flow.

    * **Helm**. Set `interval` in the [`nri-bundle` chart](https://github.com/newrelic/helm-charts/tree/master/charts/nri-bundle) the value you want.

>     * **Manifest**. Set `interval` in the Manifest configuration yaml file to the value you want. If you followed our [installation instructions](/install/kubernetes/?dropdown1=manifest), the file was called `newrelic-manifest.yaml`.

Reviewer: There is more than one interval values.

Author: I've specified `scrape interval`

Author: I've specified `scrape interval`

If you want to change the scrape interval, just add the interval value under the `newrelic-infrastructure` section. Take a look at this example of the `values-newrelic.yaml` file to see how it looks:

```yaml
global:
  licenseKey: _YOUR_NEW_RELIC_LICENSE_KEY_
  cluster: _K8S_CLUSTER_NAME_

# ... Other settings

# Configuration for newrelic-infrastructure
newrelic-infrastructure:
  # ... Other settings
  common:
    config:
      interval: 25s
```

<Callout variant="important">

>     Setting `interval` to values larger than `40s` isn't allowed.

Author: Modified the note:
Note that if you're using the [cluster explorer](/docs/kubernetes-pixie/kubernetes-integration/understand-use-data/kubernetes-cluster-explorer/#cluster-explorer-use) instead of the [Kubernetes navigator](https://docs.newrelic.com/docs/kubernetes-pixie/kubernetes-integration/understand-use-data/kubernetes-cluster-explorer/#navigator-preview), you're not allowed to set `interval` to values greater than `40s`.

---
title: Reduce data ingested
tags:
    - Integrations
    - Kubernetes integration
    - Understand and use data
metaDescription: 'New Relic''s Kubernetes integration: How to reduce the data ingested.'
redirects:
    - /docs/integrations/kubernetes-integration/understand-use-data/find-use-your-kubernetes-data
    - /docs/integrations/kubernetes-integration/understand-use-data/understand-use-data

>     - /docs/integrations/kubernetes-integration/installation/install-kubernetes-integration-using-helm/#services

Reviewer 2: @nbaenam since you deleted the governance doc, shouldn't you put a redirect in this combined doc?

`/docs/kubernetes-pixie/kubernetes-integration/advanced-configuration/data-governance`

freshnessValidatedDate: 2024-06-24
---

> If you've reviewed your [data ingestion](/docs/data-apis/manage-data/manage-data-coming-new-relic/) and want to send less data, we've got a simple solution for you. You can easily achieve this by setting up our [nri-bundle](https://github.com/newrelic/helm-charts/tree/master/charts/nri-bundle)  Helm chart. Our charts let you fine-tune your settings. This reduces the ingested data, but may make the information slightly less detailed. These setting, also allow you to focus on the most critical data points, helping you identify the most important insights.

Author: Changed!

> If you've reviewed your [data ingestion](/docs/data-apis/manage-data/manage-data-coming-new-relic/) and want to send less data, we've got a simple solution for you. You can easily achieve this by setting up our [nri-bundle](https://github.com/newrelic/helm-charts/tree/master/charts/nri-bundle)  Helm chart. Our charts let you fine-tune your settings. This reduces the ingested data, but may make the information slightly less detailed. These setting, also allow you to focus on the most critical data points, helping you identify the most important insights.

Author: Great comment, @rhetoric101, thank you! I've modified it.

> You can reduce the data in your charts and works for all installation methods: the guided install, Helm, or Manifest. Our charts can help you adjust settings to reduce the data you take in, which may result in less detailed information. These settings allow you to focus on the most important data, which can help you identify the most important insights.

Author: I've changed the entire intro.

> It's also possible to reduce data ingestion through parameters. However, the modification of the parameters will depend on the method used to install Kubernetes.

Author: Changed the intro

## Reduce data ingested by setting `global.lowDataMode` [#reduce-data-ingest]

If you're looking to cut costs, you can modify the `global.lowDataMode` key value. This will cut down on the number of metrics sent. The modification of this key will depend on the method used to install Kubernetes:

    * **Guided install** (New Relic CLI, Helm, Manifest). The installation command includes the `lowDataMode` mode by default, thanks to these parameters: `NRI_CLI_LOW_DATA_MODE`, `NRI_CLI_PROMETHEUS_AGENT_LOW_DATA_MODE`, and `NRI_CLI_LOGGING_LOW_DATA_MODE`.

    * **Helm**. Set the `global.lowDataMode` key in the [`nri-bundle` chart](https://github.com/newrelic/helm-charts/tree/master/charts/nri-bundle) to true.

>     * **Manifest**. `LowDataMod`e is only for Helm Charts, and Helms are templates applied to Manifests. When someone applies `LowDataMode` in the Helm chart, it makes a few small changes to the final Manifest. The changes in the Manifest are about specific settings, but any of them is about a parameter named `LowDataMode`.

Author: Done

>     * **Manifest**. `LowDataMod`e is only for Helm Charts, and Helms are templates applied to Manifests. When someone applies `LowDataMode` in the Helm chart, it makes a few small changes to the final Manifest. The changes in the Manifest are about specific settings, but any of them is about a parameter named `LowDataMode`.

Author: It's a little bit complicated, but it means that the parameter is in the Helm charts, but if you modify it you have to know that the change will affect to your manifest because Helm charts are templates that apply to the manifest installation.

> `lowDataMode` key affects these specific components of the [`nri-bundle` chart](https://github.com/newrelic/helm-charts/tree/master/charts/nri-bundle):

Reviewer 2: I suggested using "parameter" when talking about `lowDataMode`, but maybe it's best to call this a key? In any case, I'd suggest this revision:

The `lowDataMode` key affects these specific components of the [`nri-bundle` chart](https://github.com/newrelic/helm-charts/tree/master/charts/nri-bundle):

Author: yes, you're right. I've changed `key` with `parameter`.

    }
    ]
    ```

    This is how the previous log record would look like after enabling `lowDataMode`:

    ```json
    [
    {
        "cluster_name": "api-test",
        "container_name": "newrelic-logging",
        "namespace_name": "nrlogs",
        "pod_name": "nri-bundle-newrelic-logging-jxnbj",
        "message": "[2021/09/14 12:30:49] [ info] [engine] started (pid=1)\n",
        "stream": "stderr",
        "timestamp": 1631622649138
    }
    ]
    ```
    </Collapser>
    <Collapser
        id="nr-pixie-integration"
        title="New Relic Pixie integration"
    >

    If you enable `lowDataMode`, the `newrelic-pixie` integration performs heavier sampling on Pixie spans and collect data less often, from 10 seconds to 15 seconds. These are the `lowDataMode` settings:

    ```
    HTTP_SPAN_LIMIT: 750
    DB_SPAN_LIMIT: 250
    COLLECT_INTERVAL_SEC: 15
    ```

    You can find the default settings for these parameters and others in the [newrelic-pixie-integration](https://github.com/newrelic/newrelic-pixie-integration) Github repo.

    </Collapser>
</CollapserGroup>

## Reduce data ingested by changing the scrape interval [#scrape-interval]

> The New Relic Kubernetes integration allows changing the scrape interval at which metrics are gathered from the cluster. This lets you choose the right balance between data resolution and usage. We suggest choosing a scrape interval between 15 and 30 seconds for the best experience.

Reviewer 2: ```suggestion
The New Relic Kubernetes integration allows you to change the scrape interval when metrics are gathered from the cluster. This lets you choose the right balance between data resolution and usage. We suggest choosing a scrape interval between 15 and 30 seconds for the best experience.
```

Author: Changed.

<Callout variant="tip">
    The `lowDataMode` parameter already sets the scrape interval to 30 seconds.
</Callout>

The modification of the scrape interval will depend on the method used to install Kubernetes:

>     * **Guided install** (New Relic CLI, Helm, Manifest). You can't modify this value following our guided install flow.

Reviewer 2: @nbaenam it looks like you're using the convention of a period instead of a colon in your lists. We should probably standardize on this. I see mostly colons inside lists (across the site).

Author: @rhetoric101, you're right. I guess I sometimes forget about this. For 9 years in my former company, we used the period.

    * **Helm**. Set the scrape interval in the [`nri-bundle` chart](https://github.com/newrelic/helm-charts/tree/master/charts/nri-bundle) to the value you want.

    * **Manifest**. Set the scrape interval in the Manifest configuration yaml file to the value you want. If you followed our [installation instructions](/install/kubernetes/?dropdown1=manifest), the file was called `newrelic-manifest.yaml`.

> If you want to change the scrape interval, just add the interval value under the `newrelic-infrastructure` section. Take a look at this example of the `values-newrelic.yaml` file to see how it looks:

Reviewer 2: @nbaenam I lost the context here. If I want to change the scrape interval via the Helm chart? Why are you showing just that example and not the others?

Author: I've added this:
`Take a look at this example of the `values-newrelic.yaml` file to see how it looks for Helm:`

---
title: Reduce data ingested
tags:
    - Integrations
    - Kubernetes integration
    - Understand and use data
metaDescription: 'New Relic''s Kubernetes integration: How to reduce the data ingested.'
redirects:
    - /docs/integrations/kubernetes-integration/understand-use-data/find-use-your-kubernetes-data
    - /docs/integrations/kubernetes-integration/understand-use-data/understand-use-data
    - /docs/integrations/kubernetes-integration/installation/install-kubernetes-integration-using-helm/#services
    - /docs/kubernetes-pixie/kubernetes-integration/advanced-configuration/data-governance
freshnessValidatedDate: 2024-06-28
---

If you've reviewed your [data ingestion](/docs/data-apis/manage-data/manage-data-coming-new-relic/) and want to send less data and reduce the data ingested, we have these solutions for you:

> * Reduce data ingested by setting `global.lowDataMode`.

Author: Much better, thank you @rhetoric101!

> * Reduce data ingested by changing the scrape interval.

Author: Done.

> * Reduce data ingested by filtering namespaces.

Author: Changed!

Let's explain each option.

## Reduce data ingested by setting `global.lowDataMode` [#reduce-data-ingest]

> If you're looking to cut costs, you can modify the `global.lowDataMode` parameter value. This will cut down on the number of metrics sent. The modification of this parameter will depend on the method used to install Kubernetes:

Author: Done.

>     * **Guided install** (New Relic CLI, Helm, Manifest). The installation command includes the `lowDataMode` mode by default, thanks to these parameters: `NRI_CLI_LOW_DATA_MODE`, `NRI_CLI_PROMETHEUS_AGENT_LOW_DATA_MODE`, and `NRI_CLI_LOGGING_LOW_DATA_MODE`.

Author: I've modified to this:
`The installation command includes `lowDataMode` by default because it has these parameters:`

    * **Helm**. Set the `global.lowDataMode` parameter in the [`nri-bundle` chart](https://github.com/newrelic/helm-charts/tree/master/charts/nri-bundle) to true.

>     * **Manifest**. `LowDataMode` is only for Helm charts, and Helms charts are templates applied to manifest. When someone applies `LowDataMode` in the Helm chart, it makes a few small changes to the final manifest. The changes in the manifests are about specific settings, but any of them is about a parameter named `LowDataMode`.

Author: Done.

>     * **Manifest**. `LowDataMode` is only for Helm charts, and Helms charts are templates applied to manifest. When someone applies `LowDataMode` in the Helm chart, it makes a few small changes to the final manifest. The changes in the manifests are about specific settings, but any of them is about a parameter named `LowDataMode`.

Author: I've rewritten the sentence as follows:
`The changes in the manifests are about specific settings, but not for a parameter named `LowDataMode`.`

            "pod-template-generation": "1",
            "release": "nri-bundle"
        },
        "namespace_name": "nrlogs",
        "pod_id": "54556e3e-719c-46b5-af69-020b75d69bf1",
        "pod_name": "nri-bundle-newrelic-logging-jxnbj"
        },
        "message": "[2021/09/14 12:30:49] [ info] [engine] started (pid=1)\n",
        "plugin": {
        "source": "kubernetes",
        "type": "fluent-bit",
        "version": "1.8.1"
        },
        "stream": "stderr",
        "time": "2021-09-14T12:30:49.138824971Z",
        "timestamp": 1631622649138
    }
    ]
    ```

    This is how the previous log record would look like after enabling `lowDataMode`:

    ```json
    [
    {
        "cluster_name": "api-test",
        "container_name": "newrelic-logging",
        "namespace_name": "nrlogs",
        "pod_name": "nri-bundle-newrelic-logging-jxnbj",
        "message": "[2021/09/14 12:30:49] [ info] [engine] started (pid=1)\n",
        "stream": "stderr",
        "timestamp": 1631622649138
    }
    ]
    ```
    </Collapser>
    <Collapser
        id="nr-pixie-integration"
        title="New Relic Pixie integration"
    >

>     If you enable `lowDataMode`, the `newrelic-pixie` integration performs heavier sampling on Pixie spans and collect data less often, from 10 seconds to 15 seconds. These are the `lowDataMode` settings:

Author: Done.

    ```
    HTTP_SPAN_LIMIT: 750
    DB_SPAN_LIMIT: 250
    COLLECT_INTERVAL_SEC: 15
    ```

    You can find the default settings for these parameters and others in the [newrelic-pixie-integration](https://github.com/newrelic/newrelic-pixie-integration) Github repo.

    </Collapser>
</CollapserGroup>

## Reduce data ingested by changing the scrape interval [#scrape-interval]

The New Relic Kubernetes integration allows you to change the scrape interval when metrics are gathered from the cluster. This lets you choose the right balance between data resolution and usage. We suggest choosing a scrape interval between 15 and 30 seconds for the best experience.

<Callout variant="tip">
    The `lowDataMode` parameter already sets the scrape interval to 30 seconds.
</Callout>

> The modification of the scrape interval will depend on the method used to install Kubernetes:

Author: Done.

>     * **Guided install** (New Relic CLI, Helm, Manifest): You can't modify this value following our guided install flow.

Author: Done.

    * **Helm**: Set the scrape interval in the [`nri-bundle` chart](https://github.com/newrelic/helm-charts/tree/master/charts/nri-bundle) to the value you want.

>     * **Manifest**: Set the scrape interval in the Manifest configuration yaml file to the value you want. If you followed our [installation instructions](/install/kubernetes/?dropdown1=manifest), the file was called `newrelic-manifest.yaml`.

Author: Done.

> If you want to change the scrape interval, just add the interval value under the `newrelic-infrastructure` section. Take a look at this example of the `values-newrelic.yaml` file to see how it looks for Helm:

Author: @rhetoric101, I've modified it to this:
`If you're using Helm or manifest and want to change the scrape interval, just add the interval...`

  cluster: _K8S_CLUSTER_NAME_

# ... Other settings

# Configuration for newrelic-infrastructure
newrelic-infrastructure:
  # ... Other settings
  common:
    config:
      interval: 25s
```

<Callout variant="important">
    Note that if you're using the [cluster explorer](/docs/kubernetes-pixie/kubernetes-integration/understand-use-data/kubernetes-cluster-explorer/#cluster-explorer-use) instead of the [Kubernetes navigator](https://docs.newrelic.com/docs/kubernetes-pixie/kubernetes-integration/understand-use-data/kubernetes-cluster-explorer/#navigator-preview), you're not allowed to set `interval` to values greater than `40s`.
</Callout>

## Reduce data ingested by filtering namespaces [#filter-namespace]

You can label namespaces to filter which ones the Kubernetes integration scrapes. All namespaces scrape by default.

We use the `namespaceSelector` in the same way Kubernetes does. To include only namespaces matching a label, just change the `namespaceSelector`. Add the following to your `values-newrelic.yaml` file, under the `newrelic-infrastructure` section:

```yaml
common:
  config:
    namespaceSelector:
      matchLabels:
        key1 : "value1"
```

> 

Reviewer 2: I would add an h3 heading here like this: ### Example: Include namespaces [#include]`

Author: Done.

In this example only namespaces with the label `newrelic.com/scrape` set to `true` will be scraped:

```yaml
global:
  licenseKey: _YOUR_NEW_RELIC_LICENSE_KEY_
  cluster: _K8S_CLUSTER_NAME_

# ... Other settings 

# Configuration for newrelic-infrastructure
newrelic-infrastructure:
  # ... Other settings 
  common:
    config:
      namespaceSelector:
        matchLabels:
          newrelic.com/scrape: "true"
```

You can also use Kubernetes expressions to include or exclude namespaces using this syntax:

```yaml
common:
  config:
    namespaceSelector:
      matchExpressions:
      - {key: newrelic.com/scrape, operator: NotIn, values: ["false"]}
      - {key: key1, operator: In, values: ["value1"]}
```

<Callout variant="tip">
  The expressions under `matchExpressions` are concatenated.
</Callout>

> 

Reviewer 2: I would add an h3 heading here like this: `### Example: Exclude namespaces [#exclude]`

Author: Done although the h3 heading is as follows:
`Examples with namespaces`

In this example, namespaces with the label `newrelic.com/scrape` set to `false` will be excluded:

```yaml
global:
  licenseKey: _YOUR_NEW_RELIC_LICENSE_KEY_
  cluster: _K8S_CLUSTER_NAME_

# ... Other settings 

# Configuration for newrelic-infrastructure
newrelic-infrastructure:
  # ... Other settings
  common:
    config:
      namespaceSelector:
        matchExpressions:
        - {key: newrelic.com/scrape, operator: NotIn, values: ["false"]}
```

See a full list of settings that can be modified in the [chart's README file](https://github.com/newrelic/nri-kubernetes/tree/main/charts/newrelic-infrastructure).

### How can I find out which namespaces are excluded? [#excluded-namespaces]

The `K8sNamespace` sample shows all the namespaces within the cluster. The `nrFiltered` attribute determines whether we're going to scrape the data related to the namespace. This query  shows you which namespaces are being monitored:

```sql

> FROM K8sNamespaceSample SELECT displayName, nrFiltered WHERE clusterName = <clusterName> SINCE 2 MINUTES AGO

Reviewer 2: @nbaenam I tried to format this query according to our style guide, but GitHub is inserting spaces between each command. You could just reject this change, or try to fix it locally. The GitHub UI isn't working.

> FROM K8sNamespaceSample SELECT displayName, nrFiltered WHERE clusterName = <clusterName> SINCE 2 MINUTES AGO

Author: I see the entire query, it's working for me.

    - /docs/kubernetes-pixie/kubernetes-integration/advanced-configuration/data-governance
freshnessValidatedDate: 2024-07-09
---

If you've reviewed your [data ingestion](/docs/data-apis/manage-data/manage-data-coming-new-relic/) and want to send less data and reduce the data ingested, we have these solutions for you:

* Set `global.lowDataMode`

* Change the scrape interval

* Filter namespaces

Let's explain each option.

## Reduce data ingested by setting `global.lowDataMode` [#reduce-data-ingest]

If you're looking to cut costs, you can modify the `global.lowDataMode` parameter value. This will cut down on the number of metrics you send. The modification of this parameter will depend on the method you used to install Kubernetes:

    * **Guided install** (New Relic CLI, Helm, manifest). The installation command includes `lowDataMode` by default because it has these parameters: `NRI_CLI_LOW_DATA_MODE`, `NRI_CLI_PROMETHEUS_AGENT_LOW_DATA_MODE`, and `NRI_CLI_LOGGING_LOW_DATA_MODE`.

    * **Helm**. Set the `global.lowDataMode` parameter in the [`nri-bundle` chart](https://github.com/newrelic/helm-charts/tree/master/charts/nri-bundle) to true.

>     * **Manifest**. `LowDataMode` is only for Helm charts, and Helms charts are templates applied to manifests. When someone applies `LowDataMode` in the Helm chart, it makes a few small changes to the final manifest. The changes in the manifests are about specific settings, but not for a parameter named `LowDataMode`.

Reviewer 2: @nbaenam I'm still unable to understand this sentence. If you think more knowledgeable readers will know what this means, I'll defer.

Author: I think it's clear for K8s users, but thank you for your comment.

    </Callout>

    </Collapser>
    <Collapser
        id="exclude-namespaces"
        title={<>Exclude namespaces with the label <InlineCode>newrelic.com/scrape</InlineCode> set to <InlineCode>false</InlineCode></>}
    >

    ```yaml
    global:
    licenseKey: _YOUR_NEW_RELIC_LICENSE_KEY_
    cluster: _K8S_CLUSTER_NAME_

    # ... Other settings 

    # Configuration for newrelic-infrastructure
    newrelic-infrastructure:
    # ... Other settings
    common:
        config:
        namespaceSelector:
            matchExpressions:
            - {key: newrelic.com/scrape, operator: NotIn, values: ["false"]}
    ```
    </Collapser>
</CollapserGroup>

You can see a full list of settings that you can modify in the [chart's README file](https://github.com/newrelic/nri-kubernetes/tree/main/charts/newrelic-infrastructure).

### How can I find out which namespaces are excluded? [#excluded-namespaces]

The `K8sNamespace` sample shows all the namespaces within the cluster. The `nrFiltered` attribute determines whether we're going to scrape the data related to the namespace. This query  shows you which namespaces are being monitored:

```sql

> FROM K8sNamespaceSample SELECT displayName, nrFiltered WHERE clusterName = <clusterName> SINCE 2 MINUTES AGO

Reviewer 2: ```suggestion
FROM K8sNamespaceSample 
SELECT displayName, nrFiltered 
WHERE clusterName = <clusterName> 
SINCE 2 MINUTES AGO
```

Author: Hey @rhetoric101, the query is fine in the preview and in the file. I don't know why you see the inserted spaces.
